sdk/ai/Azure.AI.Projects/src/Custom/Agents/AgentVersionCreationOptions.cs

```diff
@@ -19,5 +19,7 @@ public partial class AgentVersionCreationOptions
     public global::Azure.AI.Projects.OpenAI.AgentDefinition Definition { get; set; }
 
     private static void DeserializeDefinitionValue(JsonProperty property, ref global::Azure.AI.Projects.OpenAI.AgentDefinition definition)
-        => CustomSerializationHelpers.DeserializeProjectOpenAIType<AgentDefinition>(property.Value, ModelSerializationExtensions.WireOptions);
+    {
+        definition = CustomSerializationHelpers.DeserializeProjectOpenAIType<AgentDefinition>(property.Value, ModelSerializationExtensions.WireOptions);
```

Comment:
Curious, how did you catch this one? It's clearly a bad bug, but we've also clearly got no current coverage for *deserializing* the creation options.

Reply:
The telemetry is showing empty section, corresponding to `AgentDefinitions`. That happens because in the telemetry we are instantiating options from [BinaryData](https://github.com/Azure/azure-sdk-for-net/blob/596a73ac34ca2941bb8bc6900f1e8b1509829bac/sdk/ai/Azure.AI.Agents/src/Telemetry/OpenTelemetryScope.cs#L183C13-L183C41) and the code above does not set the reference and hence, the definitions are remaining `null`. In further [code](https://github.com/Azure/azure-sdk-for-net/blob/596a73ac34ca2941bb8bc6900f1e8b1509829bac/sdk/ai/Azure.AI.Agents/src/Telemetry/OpenTelemetryScope.cs#L209C36-L209C58) we are setting empty value for model and other fields from definition.